### PR TITLE
Port the diffBragg diffuse halo code to the exascale/kokkos API.

### PR DESCRIPTION
**Pull request by nksauter**

### CODE REVIEW

**Comment by nksauter** [comment]
The code compiles and passes the simtbx and LS49 unit tests.  New tests need to be implemented because at present there are no tests that actually exercise the diffuse API within simtbx/kokkos, the "exascale API".  A test is also needed to compare the diffuse halo calculation of diffBragg to that of the exascale API to assure the simulated images are equal or at least close.

**Comment by nksauter** [comment]
Asking for review from @phyy-nx due to CI failure

**Review by mewall — COMMENTED** [review]
@nksauter I am able to build the code but am seeing segfaults when running on my usual gpu development machine. Will troubleshoot this in a couple of days.

File: simtbx/diffBragg/src/diffBragg_nks.cpp
Comment (dermen) [inline simtbx/diffBragg/src/diffBragg_nks.cpp:1]: @nksauter , what is this file used for ?

Comment (nksauter) [reply simtbx/diffBragg/src/diffBragg_nks.cpp:1]: File was committed by accident and has now been removed.  Thanks for noticing.

**Comment by phyy-nx** [comment]
> Asking for review from @phyy-nx due to CI failure

Looks like it was a time-out, one that happens intermittently.  We'll see if it passes this on this latest commit.

File: simtbx/kokkos/SConscript [hunk 1]
```diff
@@ -127,6 +127,18 @@ if use_sycl:
   kokkos_env.Replace(CXXFLAGS=[list(OrderedDict.fromkeys(kokkos_env['CXXFLAGS']))])
   kokkos_env.Replace(SHLINKFLAGS=[list(OrderedDict.fromkeys(kokkos_env['SHLINKFLAGS']))])
 
+if True: # diffuse code in exascale API
```
Comment (dermen) [inline simtbx/kokkos/SConscript:130]: should this if test be removed ?

File: simtbx/kokkos/kokkos_ext.cpp [hunk 2]
```diff
@@ -130,6 +174,43 @@ namespace simtbx { namespace Kokkos {
...
+        .def(init<>())
+        .add_property("enable",
+             make_getter(&simtbx::Kokkos::diffuse_api::enable,rbv()),
+             make_setter(&simtbx::Kokkos::diffuse_api::enable,dcp()),
+             "whether or not to simulate diffuse.")
+        .add_property("stencil_size",
+             make_getter(&simtbx::Kokkos::diffuse_api::stencil_size,rbv()),
+             make_setter(&simtbx::Kokkos::diffuse_api::stencil_size,dcp()),
+             "")
+        .add_property("symmetize_diffuse",
```
Comment (Trzs) [inline simtbx/kokkos/kokkos_ext.cpp:193]: symmetize or symmetrize?
I think it should be consistent between python and C++

File: simtbx/kokkos/SConscript (hunk 1 above)
Comment (nksauter) [reply simtbx/kokkos/SConscript:130]: @dermen Incoming commit explains in a comment that the block is indented to highlight code that can be removed in the future if we ever finish replacing the eigen dependency with the newer kokkoslib code.

File: simtbx/kokkos/kokkos_ext.cpp (hunk 2 above)
Comment (nksauter) [reply simtbx/kokkos/kokkos_ext.cpp:193]: @Trzs incoming commit fixes the typo.

**Comment by nksauter** [comment]
I plan to merge this branch today, then add the planned unit test at a later date.

**Pull request merged**